sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -742,6 +749,22 @@ class SparkSession private(
     DataType.fromJson(dataTypeString)
   }
 
+  /**
+   * Stops all active streaming queries
+   */
+  private def stopStreamingQueries(): Unit = {
+    if (!sparkContext.isStopped) {
+      streams.active.foreach { query =>
```

Comment:
Does it really work? IIUC the streaming queries are run asynchronously, when we reach here, some stream queries may already be failed.

Reply:
Yes its possible. Theres a [check](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecution.scala#L142) that ensures the query execution thread is still alive before stopping it. Also this is best effort attempt to stop the running queries and if the stop fails, we log and continue.